### PR TITLE
conventions: Document our existing 4.x conventions

### PR DESCRIPTION
**Pull request by smarterclayton**
Clarify resource usage and operator conventions that we have been
using over the last year.

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -89,15 +89,57 @@ metal3-io](https://github.com/metal3-io/metal3-docs/blob/master/design/bare-meta
...
+it is used to manage all elements of the cluster.
+
+* All components are managed by operators
+  * All cluster component operators are expected to deploy, reconfigure, self-heal, and report status about its health back to the cluster
+  * The cluster-version-operator (CVO) provides a minimum lifecycle for the core operators that
+    are required to get OLM running, and OLM runs all other operators
+  * The core operators are updated together and represent the minimum set of function for an OpenShift cluster
+  * Any operator that is an optional part of OpenShift or has a different lifecycle than the core OpenShift release must run under OLM
+* All configuration is exposed as API resources on cluster
+  * Immutable configuration is minimal and exposed as "status" fields of a cluster, set during installation
```
Comment (derekwaynecarr) [inline CONVENTIONS.md:111]: did you mean to say "status" fields of global configuration?  its not clear what cluster references here - a specific resource, or the entity itself.

File: CONVENTIONS.md [hunk 2]
```diff
@@ -89,15 +89,57 @@ metal3-io](https://github.com/metal3-io/metal3-docs/blob/master/design/bare-meta
...
+  * All other configuration is expected to work like normal Kube objects - changeable at any time, validated, and any failures reported via the status fields of that API
+* Every component must remain available to consumers without disruption during upgrades and reconfiguration
+  * See the next section for more details
+
+
+#### Upgrade and Reconfiguration
+
+* Every component must remain available to consumers without disruption during upgrades and reconfiguration
+  * Disruption caused by node restarts is allowed but must respect disruption policies (PodDisruptionBudget) and should optimized to reduce total disruption
```
Comment (derekwaynecarr) [inline CONVENTIONS.md:120]: nit: should be

File: CONVENTIONS.md [hunk 3]
```diff
@@ -89,15 +89,57 @@ metal3-io](https://github.com/metal3-io/metal3-docs/blob/master/design/bare-meta
...
+    * Administrators may minimize
```
Comment (derekwaynecarr) [inline CONVENTIONS.md:121]: was this the end of prior sentence or start of new?

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
just a few nits for clarity.

File: CONVENTIONS.md (hunk 1 above)
Comment (smarterclayton) [reply CONVENTIONS.md:111]: reworded

File: CONVENTIONS.md (hunk 3 above)
Comment (smarterclayton) [reply CONVENTIONS.md:121]: finished incomplete thought

**Review by derekwaynecarr — APPROVED** [review]

**Pull request merged**